src/providers/drafts.ts

```diff
@@ -0,0 +1,122 @@
+import { Language, License, VideoCategory } from '@/api/queries'
+import { createStore } from '@/store'
+import { createId } from '@/utils/createId'
+import { readFromLocalStorage } from '@/utils/localStorage'
+
+export type CommonDraftProps = {
+  id: string
+  channelId: string
+  updatedAt: string
+  seen?: boolean
+}
+
+export type Draft = VideoDraft
+
+export type DraftType = 'video'
+
+export type VideoDraft = {
+  type: 'video'
+  title?: string | null
+  description?: string | null
+  isPublic?: boolean | null
+  publishedBeforeJoystream?: string | null
+  hasMarketing?: boolean | null
+  isExplicit?: boolean | null
+  language?: Language['iso'] | null
+  category?: VideoCategory['id'] | null
+  licenseCode?: License['code'] | null
+  licenseCustomText?: License['customText'] | null
+  licenseAttribution?: License['attribution'] | null
+} & CommonDraftProps
+
+export type RawDraft = Omit<Draft, 'id' | 'updatedAt'>
+
+type DraftStoreStateV1 = {
+  allDrafts: Draft[]
+}
+type DraftStoreActions = {
+  addDraft: (draft: RawDraft, explicitId?: string) => Draft
+  updateDraft: (draftId: string, draftProps: RawDraft) => void
+  getDraft: (draftId: string) => Draft | undefined
+  getDraftsForChannel: (draftIds: string) => Draft[]
+  getUnseenDraftsForChannel: (draftIds: string) => Draft[]
```

Comment:
I don't know - I don't like it. I always thought that actions should only modify data, not getting it. In my opinion, we should use selectors instead for all of these `get` actions.  For example:
```js
const drafts = useDraftStore(({ allDrafts }) => allDrafts.filter((d) => d.channelId === activeChannelId))
```

Reply:
yeah I get the point I think we need to make separate selectors that do this but not live in the store actions